llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp

```diff
@@ -4145,6 +4156,24 @@ bool AMDGPULegalizerInfo::legalizeCTLZ_CTTZ(MachineInstr &MI,
   return true;
 }
 
+bool AMDGPULegalizerInfo::legalizeCTLZ_ZERO_UNDEF(MachineInstr &MI,
+                                                  MachineRegisterInfo &MRI,
+                                                  MachineIRBuilder &B) const {
+  Register Dst = MI.getOperand(0).getReg();
+  Register Src = MI.getOperand(1).getReg();
+  LLT SrcTy = MRI.getType(Src);
+  TypeSize NumBits = SrcTy.getSizeInBits();
+
+  assert(NumBits < 32u);
+
+  auto ShiftAmt = B.buildConstant(S32, 32u - NumBits);
+  auto Extend = B.buildAnyExt(S32, {Src}).getReg(0u);
+  auto Shift = B.buildLShr(S32, {Extend}, ShiftAmt).getReg(0u);
+  B.buildInstr(AMDGPU::G_AMDGPU_FFBH_U32, {Dst}, {Shift});
```

Comment:
This isn't taking advantage of the 0-is-1 behavior, so this is really just another flavor of CTLZ_ZERO_UNDEF? This can be done in the generic LegalizerHelper?

Reply:
`LegalizerHelper` uses compare & select to lower `CTLZ_ZERO_UNDEF` so we end up with a min and subtract.